mod/consensus/pkg/cometbft/service/genesis.go

```
@@ -44,13 +62,188 @@
+func validateDeposits(deposits []types.Deposit) error {
+	if len(deposits) == 0 {
+		return errors.New("at least one deposit is required")
+	}
+
+	seenPubkeys := make(map[string]bool)
+
+	for i, deposit := range deposits {
+		depositIndex := deposit.GetIndex()
+		//#nosec:G701 // realistically fine in practice.
+		// Validate index matches position
+		if int64(depositIndex) != int64(i) {
```

golangci-lint (1.62.2)

[high] G115: integer overflow conversion uint64 -> int64

(gosec)